From b7051bc715c0f8943c281418b22625f5d622f5cf Mon Sep 17 00:00:00 2001
From: Neal Gompa <ngompa@datto.com>
Date: Fri, 30 Mar 2018 17:01:14 -0400
Subject: [PATCH 14/18] Remove default values for keypair and security names
 for EC2LatentBuildSlave

This is a simpler/trivial adaptation of 68a9267d5fff06e0ff7c6ea8a82ab66fcf6a359c in buildbot 0.9.x,
originally authored by Vladimir Rutsky <vladimir@rutsky.org>.

There are no test adaptations for this, as ultimate behavior has not yet changed.

Signed-off-by: Neal Gompa <ngompa@datto.com>
---
 master/buildbot/buildslave/ec2.py | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/master/buildbot/buildslave/ec2.py b/master/buildbot/buildslave/ec2.py
index 9f24df51d..780c4ef05 100644
--- a/master/buildbot/buildslave/ec2.py
+++ b/master/buildbot/buildslave/ec2.py
@@ -61,8 +61,8 @@ class EC2LatentBuildSlave(AbstractLatentBuildSlave):
                  valid_ami_owners=None, valid_ami_location_regex=None,
                  elastic_ip=None, identifier=None, secret_identifier=None,
                  aws_id_file_path=None, user_data=None, region=None,
-                 keypair_name='latent_buildbot_slave',
-                 security_name='latent_buildbot_slave',
+                 keypair_name=None,
+                 security_name=None,
                  max_builds=None, notify_on_missing=[], missing_timeout=60 * 20,
                  build_wait_timeout=60 * 10, properties={}, locks=None,
                  spot_instance=False, max_spot_price=1.6, volumes=[],
@@ -95,10 +95,15 @@ class EC2LatentBuildSlave(AbstractLatentBuildSlave):
             else:
                 # verify that regex will compile
                 re.compile(valid_ami_location_regex)
+        if keypair_name is None:
+            keypair_name = 'latent_buildbot_slave'
+            log.msg('Using default keypair name, since none is set')
+        if security_name is None:
+            security_name = 'latent_buildbot_slave'
+            log.msg('Using default keypair name, since none is set')
         if spot_instance and price_multiplier is None and max_spot_price is None:
             raise ValueError('You must provide either one, or both, of '
                              'price_multiplier or max_spot_price')
-
         self.valid_ami_owners = valid_ami_owners
         self.valid_ami_location_regex = valid_ami_location_regex
         self.instance_type = instance_type
-- 
2.14.3

